timeinterval/repeating_interval.go

```diff
@@ -0,0 +1,164 @@
+package timeinterval
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+)
+
+// RepeatingInterval describes an interval with recurring events distributed evenly by a fixed duration.
+// The interval can be bounded by either:
+// a fixed startsAt and endsAt
+// or by a fixed startsAt with a fixed number of Repetitions from which the endsAt will be derived.
+// or by a fixed endsAt with a fixed number of Repetitions from which the startsAt will be derived.
+type RepeatingInterval struct {
```

Comment:
```suggestion
type Repeating struct {
```

For the user it will be `timeinterval.Repeating` which is better without the stutter.